CRM/Utils/Check/Component/Cleanurls.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+class CRM_Utils_Check_Component_Cleanurls extends CRM_Utils_Check_Component {
+
+  /**
+   * For sites running in WordPress, make sure clean URLs are properly set in settings file.
+   *
+   * @return CRM_Utils_Check_Message[]
+   */
+  public static function checkWpCleanurls() {
+    $config = CRM_Core_Config::singleton();
+    //Right now this config issue largely affects Wordpress only. This check could be applied to other CMS' but for now it's just WP
+    if ($config->userFramework != 'WordPress') {
```

Comment:
The test fail is a style issue where it doesn't like the indentation on line 96, but regarding cms-specific, a preferred way would be:

1. Put all of the code below in this function here instead into CRM_Utils_System_Wordpress in a function called checkCleanurls()
2. In CRM_Utils_System_Base, make a function checkCleanurls() that just returns `[]`
3. And then here this becomes a one-line function which will automatically do the right thing for all cmses, and is easy to update independently for other cmses by making overrides in their own CRM_Utils_System_XXX.

```php
public static function checkCleanurls() {
  return CRM_Core_Config::singleton()->userSystem->checkCleanurls();
}
```

Reply:
Thanks! I just followed CRM_Utils_Check_Component_Cms but this makes sense as a way to organize it.

Reply:
Oh that one could probably be cleaned up too someday (grin).